### PR TITLE
Carbonmark data tables

### LINKED ISSUE #1455: [carbon-dashboard] Develop new chart type: Paginated table
Good example: Detailed list of KlimaDAO retirements

### PR DESCRIPTION
**Pull request by biwano**
## Description

- Introduces a generic Table component
- Detailed list of KlimaDAO Retirements

To see the table click on "Retirement Trends" > "By Token" > Scroll down

There is a strange prettier behaviour in one of the files introduced by this PR. See: https://github.com/KlimaDAO/klimadao/issues/1528

## Related Ticket

Resolves #1455

## Changes

| Before  | After  |
|---------|--------|
|NA |imageimage
|

## Checklist

- [X] I have run `npm run build-all` without errors
- [X] I have formatted JS and TS files with `npm run format-all`

### CODE REVIEW

File: carbonmark-data/components/charts/helpers/DataTable/Pagination/styles.module.scss [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+.paginationIcon {
+  display: flex;
+  width: 2.4rem;
+  height: 2.4rem;
+  justify-content: center;
+  align-items: center;
+  border-radius: 0.4rem;
+  background-color: var(--text-color-06);
+  color: white;
```
Comment (sprrwhwk) [inline carbonmark-data/components/charts/helpers/DataTable/Pagination/styles.module.scss:25]:
```suggestion
  color: white;
  cursor: pointer;
```

File: carbonmark-data/components/cards/KlimaDAORetirementsCard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import ChartCard, { CardProps } from "../ChartCard";
+
+export default function KlimaDAORetirementsCard(props: CardProps) {
+  /* @ts-expect-error async Server component */
+  const chart = <DataTable configurationKey="klimaRetirements"></DataTable>;
+
+  return (
+    <ChartCard
+      {...props}
+      title={t`Detailed list of KlimaDAO Retirements`}
```
Comment (sprrwhwk) [inline carbonmark-data/components/cards/KlimaDAORetirementsCard/index.tsx:12]:
```suggestion
      title={t`Detailed list of KlimaDAO retirements`}
```

File: carbonmark-data/components/charts/helpers/DataTable/ClientWrapper/index.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+"use client";
+import Pagination from "components/charts/helpers/DataTable/Pagination";
+import Table from "components/charts/helpers/DataTable/Table";
+import { useState } from "react";
+import { ConfigurationKey } from "../configurations";
+import styles from "./styles.module.scss";
+
+export default function ClientWrapper(props: {
```
Comment (sprrwhwk) [inline carbonmark-data/components/charts/helpers/DataTable/ClientWrapper/index.tsx:8]: I don't like the name of this component; it is too abstract and generic IMO

File: carbonmark-data/components/charts/helpers/DataTable/configurations/KlimaRetirementsTableConfiguration.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        header: t`Pool`,
+        cellStyle: layout.textCenter,
+        dataKey: "token",
+        formatter: (x: string | number) => x,
+      },
+      retirement_date: {
+        header: t`Date`,
+        cellStyle: layout.textRight,
+        dataKey: "retirement_date",
+        formatter: helpers.formatDateAsDaysShort(locale),
```
Comment (sprrwhwk) [inline carbonmark-data/components/charts/helpers/DataTable/configurations/KlimaRetirementsTableConfiguration.tsx:49]: This format doesn't match the figma format of yyyy-MM-dd

File: carbonmark-data/components/charts/helpers/DataTable/configurations/KlimaRetirementsTableConfiguration.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        cellStyle: layout.textLeft,
+        dataKey: "beneficiary",
+        formatter: (x: string | number) => x,
+      },
+      project_id: {
+        header: t`Project`,
+        cellStyle: layout.textCenter,
+        dataKey: "project_id",
+        formatter: (projectId: string) => (
+          <VerraProjectLink projectId={projectId}></VerraProjectLink>
```
Comment (sprrwhwk) [inline carbonmark-data/components/charts/helpers/DataTable/configurations/KlimaRetirementsTableConfiguration.tsx:36]: These links are underlined in figma

**Review by sprrwhwk — CHANGES_REQUESTED** [review]
Works as expected, some minor styling issues though. Great work 👍 🎉

File: carbonmark-data/components/charts/helpers/DataTable/Pagination/index.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      <div className={styles.paginationCounter}>
+        {props.page + 1} / {props.pages_count}
+      </div>
+
+      {props.page < props.pages_count - 1 && (
+        <div className={styles.paginationButtons}>
+          <div
+            className={styles.paginationIcon}
+            onClick={() => props.onPageChange(props.page + 1)}
+          >
```
Comment (sprrwhwk) [inline carbonmark-data/components/charts/helpers/DataTable/Pagination/index.tsx:46]: We can add the role attribute for better a11y

```suggestion
          <div
            className={styles.paginationIcon}
            onClick={() => props.onPageChange(props.page + 1)}
            role="button"
          >
```

File: carbonmark-data/components/cards/KlimaDAORetirementsCard/index.tsx (hunk 2 above)
Comment (biwano) [reply carbonmark-data/components/cards/KlimaDAORetirementsCard/index.tsx:12]: The 'R' is capitalized in figma WDYT @jabby09 ?

File: carbonmark-data/components/charts/helpers/DataTable/configurations/KlimaRetirementsTableConfiguration.tsx (hunk 4 above)
Comment (biwano) [reply carbonmark-data/components/charts/helpers/DataTable/configurations/KlimaRetirementsTableConfiguration.tsx:49]: Yeah I know, but I am waiting for @theprofessora advice on enforcing figma formats or using the browser localisation capabilities. See: https://github.com/KlimaDAO/klimadao/issues/1504 (In france for instance we use DD-MM-YYYY and the american format gives us headaches)

File: carbonmark-data/components/charts/helpers/DataTable/configurations/KlimaRetirementsTableConfiguration.tsx (hunk 5 above)
Comment (biwano) [reply carbonmark-data/components/charts/helpers/DataTable/configurations/KlimaRetirementsTableConfiguration.tsx:36]: Done in f1648082d

File: carbonmark-data/components/charts/helpers/DataTable/ClientWrapper/index.tsx (hunk 3 above)
Comment (biwano) [reply carbonmark-data/components/charts/helpers/DataTable/ClientWrapper/index.tsx]: done in 03a74f38b

**Review by wowselim — APPROVED** [review]
👌

**Review by sprrwhwk — APPROVED** [review]

**Pull request merged**